libvast/test/validate.cpp

```diff
@@ -35,6 +35,12 @@ auto test_layout = record_type{
     }}
   };
 
+auto test_layout2 = record_type{
+  {"struct", record_type{
+    {"foo", string_type{}, {"required"}},
```

Comment:
```suggestion
    {"foo", type{string_type{}, {"required"}}},
```

And then you no longer need the modification in type.hpp at all.

Reply:
Almost, missing one more set of `{}` ;)